airbyte-webapp/src/pages/SettingsPage/pages/DbMigrationPage/components/MigrationComponent.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useEffect, useState } from "react";
+import { useFetcher } from "rest-hooks";
+import DbMigrationResource, {
+  DbMigrationInfoItem,
+} from "core/resources/DbMigration";
+import useDbMigration from "components/hooks/services/useDbMigrationHook";
+import { useAsyncFn } from "react-use";
+import { Block, Title } from "../../ConnectorsPage/components/PageComponents";
+import { LoadingButton } from "components";
+import { FormattedMessage } from "react-intl";
+import MigrationTableView from "./MigrationTableView";
+
+type DbMigrationComponentProps = {
+  databaseTitle: string;
+  databaseIdentifier: string;
+};
+
+const MigrationComponent: React.FC<DbMigrationComponentProps> = ({
+  databaseTitle,
+  databaseIdentifier,
+}) => {
+  const { info, migrate } = useDbMigration();
+
+  const [migrations, setMigrations] = useState<Array<DbMigrationInfoItem>>([]);
+  const [hasPendingMigration, setHasPendingMigration] = useState(false);
+
+  useEffect(() => {
+    info(databaseIdentifier).then((result) => setMigrations(result.migrations));
+  }, []);
+
+  useEffect(() => {
+    const hasPending = migrations.some(
+      (migration) => migration.migrationState === "Pending"
```

Comment:
Bad practice. Just use `useMemo` if you want to have new state or filter object.

```
const hasPending = useMemo(() => migrations.some((migration) => migration.migrationState === "Pending") , [migrations]);```